aten/src/ATen/native/cpu/SumKernel.cpp

```diff
@@ -11,6 +11,24 @@ namespace at {
 namespace native {
 namespace {
 
+// These macros help decrease duplicate code for AVX2 & AVX512
+#if defined(CPU_CAPABILITY_AVX2) && !defined(_MSC_VER)
+#define outer_sum_bf16_vec  __m128i
+#define inner_sum_bf16_vec __m256i
+#define inner_sum_float_vec_add _mm256_add_ps
+#define float_vec __m256
+#define load_outer_sum_bf16_vec _mm_loadu_si128
+#define load_float_vec _mm256_loadu_si256
+#elif defined(CPU_CAPABILITY_AVX512) && !defined(_MSC_VER)
+#define outer_sum_bf16_vec  __m256i
+#define inner_sum_bf16_vec __m512i
+#define inner_sum_float_vec_add _mm512_add_ps
+#define float_vec __m512
+#define load_outer_sum_bf16_vec _mm256_loadu_si256
+#define load_float_vec _mm512_loadu_si512
+#endif
```

Comment:
Hello @peterbell10, if possible, whenever you can, please share your opinion on these macros (if their names make sense), which aim to reduce redundant code when AVX512 support would be added, so that `InnerSumCastLoadPolicy` & `InnerSumCastLoadPolicy` wouldn't have duplicated code for AVX512. Thanks!

Reply:
To be honest I find this macro-based solution ugly and think it obfuscating the meaning of the code. How about having two helper functions in the `cpu/vec` folder to deal with loading into `Vectorized<float>`
```cpp
// First helper loads one Vectorized<float>
#if defined(CPU_CAPABILITY_AVX2) && !defined(_MSC_VER)
void load_fp32_from_bf16(const c10::BFloat16* data, Vectorized<float> &out) {
    auto bf = _mm_loadu_si128(data);
    cvtbf16_fp32(bf, out.values);
}
#elif defined(CPU_CAPABILITY_AVX512) && !defined(_MSC_VER)
void load_fp32_from_bf16(const c10::BFloat16* data, Vectorized<float> &out) {
    auto bf = _mm_loadu_si128(data);
    cvtbf16_fp32(bf, out.values);
}
#endif

// Second helper loads two Vectorized<float>
#if (defined(CPU_CAPABILITY_AVX2) || defined(CPU_CAPABILITY_AVX512)) && !defined(_MSC_VER)
void load_fp32_from_bf16(const c10::BFloat16* data, Vectorized<float> &out1, Vectorized<float> &out2) {
  auto bf = Vectorized<c10::BFloat16>::load(data);
  cvtbf16_fp32(bf.values, out1.values, out2.values);
}
#endif
```

Reply:
I'll try to add that in #61082, so in theory you only need to add new overloads for AVX512 in this PR.

Reply:
Thanks a lot for the feedback, @peterbell10! I completely agree.